src/routes/safe/components/Apps/ManageApps.tsx

```diff
@@ -66,44 +129,28 @@ const ManageApps = ({ appList, onAppAdded, onAppToggle }) => {
 
   const cleanAppInfo = () => setAppInfo(APP_INFO)
 
-  const safeAppValidator = async (value) => {
-    const appInfo = await getAppInfoFromUrl(value)
-
-    if (appInfo.error) {
-      setAppInfo(APP_INFO)
-      return 'This is not a valid Safe app.'
+  const onFormStatusChange = async ({ pristine, valid, validating, values, errors }) => {
+    if (!pristine) {
+      setIsSubmitDisabled(validating || !valid)
     }
 
-    setAppInfo({ ...appInfo })
-  }
-
-  const uniqueAppValidator = (value) => {
-    const exists = appList.find((a) => {
-      try {
-        const currentUrl = new URL(a.url)
-        const newUrl = new URL(value)
-        return currentUrl.href === newUrl.href
-      } catch (error) {
-        return 'There was a problem trying to validate the URL existence.'
-      }
-    })
-    return exists ? 'This app is already registered.' : undefined
-  }
-
-  const onFormStatusChange = ({ pristine, valid, validating }) => {
-    if (!pristine) {
-      setIsSubmitDisabled(validating || !valid || appInfo.error)
+    if (validating) {
+      return
     }
-  }
 
-  const customRequiredValidator = (value) => {
-    if (!value || !value.length) {
+    if (!values.appUrl || !values.appUrl.length || errors.appUrl) {
       setAppInfo(APP_INFO)
-      return 'Required'
+      return
     }
+
+    const url = await getUrlFromFormValue(values.appUrl)
+    const appInfo = await getAppInfoFromUrl(url)
+    setAppInfo({ ...appInfo })
   }
 
-  const getAddAppForm = () => {
+  const safeAppValidator = curriedSafeAppValidator(appList)
+
+  const getForm = () => {
```

Comment:
Can we extract this to a separate component? It will return new object everytime which could possibly hurt the performance

Reply:
good catch!